### PR TITLE
fix(storefront): commit cart quantity changes once the edit is finished

### LINKED ISSUE #19871: a11y: Quantity change causes auto submit
### Technical TODO

When changing the quantity in offcanvas cart, cart and checkout, the form auto submits to update the cart data. This violates the following WCAG criteria:

- 3.2.2 On Input: 
  - https://www.w3.org/TR/WCAG22/#on-input
  - https://www.w3.org/WAI/WCAG22/Techniques/failures/F36
- 3.2.5 Change on Request: 
  - https://www.w3.org/TR/WCAG22/#change-on-request
  - https://www.w3.org/WAI/WCAG22/Understanding/change-on-request

To avoid that:

- a visible submit button (like "update quantity") should be included
- the form should be submittable via "Enter" key, when focus is on the quantity input field

**Steps to reproduce**

- add product to cart
- open the offcanvas cart
  - change the quantity either by using the buttons or direct input
- navigate to the cart page
  - change the quantity either by using the buttons or direct input
- navigate to checkout confirm page
  - change the quantity either by using the buttons or direct input (this has to be enabled)

### PR DESCRIPTION
**Pull request by mstegmeyer**
### 1. Why is this change necessary?

`input[type=number]` emits a `change` event on every arrow key press. The cart quantity form submits on `change`, so every single arrow press reloaded the page while a keyboard or screen reader user was still choosing a value. The 800ms debounce only hides this for someone stepping fast; a screen reader user waits for each value to be announced before pressing again, so for them every press reloads.

This is also why #14015 kept coming back as focus loss despite the persistent focus handler added during the accessibility push: focus restoration treats the symptom, the reload is the cause.

For WCAG 3.2.2 the control additionally never advised the user that it applies on input, which is the escape clause that criterion offers (technique G13).

### 2. What does this change do, exactly?

`QuantitySelectorPlugin` stops treating every `change` as a finished edit. A value the user is still working on is held back and passed on when they signal they are done: leaving the input, or pressing `Enter`. That change carries `detail.submitImmediately`, and both form handlers, `FormAutoSubmitPlugin` on the cart and confirm page and `OffCanvasCartPlugin` in the off-canvas, apply it right away through their usual submission path, so AJAX, redirect parameters and extension hooks stay where they are.

Steps made with `[+]`/`[-]` or the input's native spinner keep the form's 800ms delay, which bundles repeated presses. A press is remembered from `pointerdown`, so browsers that do not focus a clicked button bundle a typed value with the step as well instead of sending both. The form's `submit` becomes the baseline the selector compares against, so nothing goes out twice, and stepping back to the rendered quantity sends nothing at all.

`OffCanvasCartPlugin` listens on the form, where a withheld event does not arrive, keeps only the response of its latest request, and restores focus to where the user actually is. `Debouncer.debounce()` returns a function with `cancel()` and `flush()` for that immediate path.

The cart quantity `<legend>` gained a snippet advising that changing the quantity updates the cart. A `<legend>` is announced before the fieldset contents, which is where 3.2.2 wants the advisory.

This deliberately does not add the visible "update quantity" button from the issue. That is driven by 3.2.5 Change on Request, which is Level AAA and therefore not an EAA obligation, it changes conversion-relevant cart UX, and it would only make sense applied consistently to the other auto-submit forms (payment, shipping, shipping cost precalc, account newsletter). Worth a product decision rather than an issue fix.

### 3. Describe each step to reproduce the issue or behaviour.

- Add a product to the cart and go to the cart page
- Focus the quantity input and press the up arrow, pausing roughly a second between presses
- Before: the page reloads after each press. After: the value steps freely and the cart updates as soon as you leave the field or press `Enter`
- Type a value and click `[+]` in Safari: one request with the stepped value

Same in the off-canvas cart and on the checkout confirm page.

### 4. Please link to the relevant issues (if any).

- fixes #19871
- relates #14015

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [ ] I have written or adjusted the documentation and agent skills according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

### CODE REVIEW

**Review by daniel3010 — APPROVED** [review]

**Review by cyl3x — APPROVED** [review]

File: src/Storefront/Resources/app/storefront/src/plugin/quantity-selector/quantity-selector.plugin.js [hunk 1]
```diff
@@ -92,11 +175,7 @@ export default class QuantitySelectorPlugin extends Plugin {
         const event = new Event('change', { bubbles: true, cancelable: false });
```
Comment (tobiasberge) [inline src/Storefront/Resources/app/storefront/src/plugin/quantity-selector/quantity-selector.plugin.js]: I have two questions:

1. Would it maybe be enough to only toggle the Event `bubbles: true/false` via a new option? From my understanding it shouldn't trigger the change event when set to false. But I haven't tested this manually.

2. This change/PR also effects the static cart (which reloads) and not only the OffCanvas, right? The static cart uses `FormAutoSubmitPlugin` which then auto reloads. Would it work to use the `changeTriggerSelectors` option and explicitly give it only the selectors (e.g. the actual quantity input) that should trigger a change - and ignore everything else (the +/- buttons).

If that doesn't work or I don't see the full context - totally fine. I was just wondering if we can make the change smaller or use existing options.

**Review by tobiasberge — COMMENTED** [review]
This is a very welcome a11y change. 👍 The current behaviour is indeed annoying with the reload. I just had some questions.

File: src/Storefront/Resources/app/storefront/src/plugin/quantity-selector/quantity-selector.plugin.js (hunk 1 above)
Comment (mstegmeyer) [reply src/Storefront/Resources/app/storefront/src/plugin/quantity-selector/quantity-selector.plugin.js]: Tested both, neither works sadly.

**1.** Line 175 is our own event, the one we *want* the form to see. What we're
blocking is the browser's, fired on every arrow key press:

```
FORM saw change: target=q bubbles=true isTrusted=true
```

`isTrusted=true`, so there's no constructor for us to pass `bubbles: false` into.

**2.** Yes, static cart is the main case. Cart and confirm do the full reload,
OffCanvas goes through ajax.

`changeTriggerSelectors` filters on `event.target` though, and everything here
targets the input, including those arrow key events. So whitelisting the quantity
input keeps exactly what we need to block. It also wouldn't skip the +/- buttons,
`_triggerChange()` dispatches on the input first and only then on the button.

What we need is "is the user done editing" and that option only answers "which
element".

(I did try owning the arrow keys in `keydown` instead, but typing still fires a
native change on blur, so `stopPropagation()` comes back anyway.)

Comment (tobiasberge) [reply src/Storefront/Resources/app/storefront/src/plugin/quantity-selector/quantity-selector.plugin.js]: Ok, thanks for taking a look. Was just wondering if we can make it more "simple" since this now adds more complexity. If this does not work, I'm fine with the current solution.

**Review by tobiasberge — APPROVED** [review]

**Review by untilu29 — APPROVED** [review]
